lib/Serialization/SILFormat.h

```diff
@@ -404,17 +406,36 @@ namespace sil_block {
   using SILOneOperandLayout = BCRecordLayout<
     SIL_ONE_OPERAND,
     SILInstOpCodeField,
-    BCFixed<4>,          // Optional attributes
+    BCFixed<2>,          // Optional attributes
     TypeIDField,
     SILTypeCategoryField,
     ValueIDField
   >;
 
+  using SILOneOperandExtraAttributeLayout = BCRecordLayout<
+    SIL_ONE_OPERAND_EXTRA_ATTR,
+    SILInstOpCodeField,
+    BCFixed<5>, // Optional attributes
```

Comment:
I don't think saving a few bits here is worth the complexity of an extra record, but you can overrule me on that.

Reply:
I was wondering where to draw the line, then I went through the process of figuring out how many bits each instruction was using, then finally decided to do this, more to reduce cognitive load than as an optimization.
- I think we should be able to add SIL attributes without wondering whether it's noticeably affecting the overall module size. This avoids the issue entirely.
- This makes it obvious which instructions are using extra attributes. There are only a few. That makes it super easy to verify through inspection that we have the right number of bits and which instructions are using them.

I don't know if that "simplification" plus the time saving is worth the extra switch case in deserialization. But I obviously like it better this way or wouldn't have done the work.